### PR TITLE
[WIP] Draft v3 root changes

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

#### Ticket Link

Fixes

#### Release Note

```release-note

```

### CODE REVIEW

**Comment by asraa** [comment]
- [x] Reduce root/target expiration to 4 months rather than [6](https://github.com/sigstore/root-signing/blob/03e99f8018ee1189ef9984f9ef09569808a810f5/cmd/tuf/app/init.go#L95)
- [x]  Rotate out one key-holder TBD
- [ ]  Create a test setup with >1 YubiKey
- [x]  Add custom metadata indicated usage/lifetime/version of the target metadata for the Rekor/Fulcio/CT key
- [x]  Add a revocation delegation @haydentherapper
- [ ]  Enable consistent snapshots (https://github.com/sigstore/root-signing/issues/80)
- [ ]  (Optional?): Start including the HSM certs into the root metadata
Changes in snapshot/targets:
- [ ]  Snapshot expiration to 5 weeks so re-signing every 3
- [ ]  Timestamp expiration to 7 days so re-signing w/ automatic merge

**Comment by asraa** [comment]
Custom metadata added as:
```
"targets": {
      "artifact.pub": {
        "custom": {
          "sigstore": {
            "status": "Active",
            "usage": "Cosign"
          }
        },
        "hashes": {
          "sha256": "59ebf97a9850aecec4bc39c1f5c1dc46e6490a6b5fd2a6cacdcac0c3a6fc4cbf",
          "sha512": "308fd1d1d95d7f80aa33b837795251cc3e886792982275e062409e13e4e236ffc34d676682aa96fdc751414de99c864bf132dde71581fa651c6343905e3bf988"
        },
        "length": 177
      },
}
```

**Comment by dlorenc** [comment]
What's left here? Need any help?

**Comment by asraa** [comment]
> What's left here? Need any help?

Hey! So far key rotation worked and I got blocked on delegation "refreshing" https://github.com/theupdateframework/go-tuf/issues/239

It could be a simple fix that I can work around in the wrapping TUF API code. I'll be able to try that today and do another dry run. Would love help for a very quick dry run later this week! @priyawadhwa since you likely have a key too.

**Comment by priyawadhwa** [comment]
Yup, I definitely had one! Will try to find it 😅

**Comment by trishankatdatadog** [comment]
> Hey! So far key rotation worked and I got blocked on delegation "refreshing" [theupdateframework/go-tuf#239](https://github.com/theupdateframework/go-tuf/issues/239)

Let us know how you need help here. @ethan-lowman-dd

File: scripts/step-1.sh [hunk 1]
```diff
@@ -16,8 +16,20 @@ export REPO=$(pwd)/ceremony/$CEREMONY_DATE
...
+git status
+
+# Ask user to insert key 
+read -n1 -r -s -p "Insert your Yubikey, then press any key to continue..." 
+
 # Add the key!
 ./tuf add-key -repository $REPO
+
+# Ask user to remove key (and replace with SSH security key)
+read -n1 -r -s -p "Remove your Yubikey, then press any key to continue..."
```
Comment (asraa) [inline scripts/step-1.sh:31]: @haydentherapper are you sure you didn't see this prompt? 
https://github.com/asraa/test-sigstore-root/issues/80#issuecomment-1086201454

maybe it's possible tapping the yubikey had pressed a key? Maybe I'll make it an explicit type Y?

**Pull request closed without merging**